internal/services/groups/groups_data_source.go

```diff
@@ -139,10 +158,13 @@ func groupsDataSourceRead(ctx context.Context, d *schema.ResourceData, meta inte
 		if group.DisplayName == nil {
 			return tf.ErrorDiagF(errors.New("API returned group with nil displayName"), "Bad API response")
 		}
-
 		newObjectIds = append(newObjectIds, *group.ID)
 		newDisplayNames = append(newDisplayNames, *group.DisplayName)
 	}
+	// Check if securityEnabled/mailEnabled has caused the number of returned groups to be 0
+	if len(newObjectIds) == 0 && (securityEnabled || mailEnabled) {
+		return tf.ErrorDiagF(errors.New("No groups found with a filter flag applied"), "Unexpected Number of groups returned")
+	}
```

Comment:
Not sure what scenario this is intended to catch as this would already have errored in the main if-else block above?